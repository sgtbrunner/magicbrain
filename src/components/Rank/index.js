import Rank from './rank.component';

export default Rank;
