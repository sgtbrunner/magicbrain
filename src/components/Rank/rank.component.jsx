import React from 'react';
import PropTypes from 'prop-types';

const Rank = ({ name, entries }) => (
  <div>
    <div className="white f4">{`Hello ${name}, your current entry count is...`}</div>
    <div className="white f2" data-testid="user-entries">
      {entries}
    </div>
  </div>
);

Rank.propTypes = {
  name: PropTypes.string.isRequired,
  entries: PropTypes.number.isRequired,
};

export default Rank;
