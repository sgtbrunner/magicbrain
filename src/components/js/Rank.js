import React from 'react';

const Rank = () => {
  return (
    <div>
      <div className='white f4'>
        {`Hello USER, your current entry count is...`}
      </div>
      <div className='white f2'>
        #?
      </div>
    </div>
  );
}

export default Rank;