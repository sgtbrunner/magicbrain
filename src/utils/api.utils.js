const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
