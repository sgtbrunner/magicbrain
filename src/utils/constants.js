export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
