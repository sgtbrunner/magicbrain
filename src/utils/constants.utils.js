const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
