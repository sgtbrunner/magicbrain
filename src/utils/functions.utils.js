export const getUserFromLocalStorage = () => localStorage.getItem('user');
