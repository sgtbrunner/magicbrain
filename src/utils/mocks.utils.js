export const USER = {
  id: 3,
  name: 'test',
  email: 'test@test.com',
  entries: 7,
};

export const OUTPUTS = {
  outputs: [
    {
      data: {
        regions: [
          {
            region_info: {
              bounding_box: {
                top_row: 0.5,
                left_col: 0.5,
                bottom_row: 0.5,
                right_col: 0.5,
              },
            },
          },
        ],
      },
    },
  ],
};

export const CREDENTIAL_ERROR = {
  error: 'You have entered an invalid username and/or password',
};

export const SERVER_ERROR = {
  error: 'Something went wrong. Please try again later.',
};
