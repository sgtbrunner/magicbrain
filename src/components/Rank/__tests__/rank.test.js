import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import Rank from '..';

const mountRankComponent = (props) => mountComponent({ component: Rank, props });

const USER_ENTRIES = 'user-entries';

describe('Rank component', () => {
  it('should match snapshot', () => {
    const { container } = mountRankComponent({ name: '', entries: 0 });

    expect(container).toMatchSnapshot();
  });

  it(`should render thirteen(13) entries rank info for user 'test'`, () => {
    const name = 'test';
    const entries = 13;
    mountRankComponent({ name, entries });

    const nameText = screen.getByText(`Hello ${name}, your current entry count is...`);
    expect(nameText).toBeInTheDocument();

    const entriesText = screen.getByTestId(USER_ENTRIES);
    expect(entriesText).toHaveTextContent(entries);
  });

  it(`should render thirteen(23) entries rank info for user 'admin'`, () => {
    const name = 'admin';
    const entries = 23;
    mountRankComponent({ name, entries });

    const nameText = screen.getByText(`Hello ${name}, your current entry count is...`);
    expect(nameText).toBeInTheDocument();

    const entriesText = screen.getByTestId(USER_ENTRIES);
    expect(entriesText).toHaveTextContent(entries);
  });
});
