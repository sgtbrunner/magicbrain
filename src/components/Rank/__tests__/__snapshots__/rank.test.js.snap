// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rank component should match snapshot 1`] = `
<div>
  <div>
    <div
      class="white f4"
    >
      Hello , your current entry count is...
    </div>
    <div
      class="white f2"
      data-testid="user-entries"
    >
      0
    </div>
  </div>
</div>
`;
